flutter/lib/src/widgets_binding_observer.dart

```diff
@@ -0,0 +1,155 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:sentry_flutter/sentry_flutter.dart';
+
+/// This is a `WidgetsBindingObserver` which can observe some events of a
+/// Flutter application.
+/// These are for example events related to its lifecycle, accessibility
+/// features, display features and more.
+///
+/// This class calls `WidgetsBinding.instance` but we don't need to call
+/// `WidgetsFlutterBinding.ensureInitialized()` because we can only add this
+/// class to an instance of `WidgetsBinding`.
+///
+/// See also:
+///   - [WidgetsBindingObserver](https://api.flutter.dev/flutter/widgets/WidgetsBindingObserver-class.html)
+class SentryWidgetsBindingObserver with WidgetsBindingObserver {
+  SentryWidgetsBindingObserver({Hub hub}) {
+    this.hub = hub ?? HubAdapter();
+  }
+
+  Hub hub;
+
+  /// This method records lifecycle events.
+  /// It tries to mimic the behavior of ActivityBreadcrumbsIntegration of Sentry
+  /// Android for lifecycle events.
+  ///
+  /// On Android and iOS this records lifecycle event breadcrumbs which loosely
+  /// correspond to their respectiv platforms lifecycle events.
+  ///
+  /// Does this need to be tracked? The nativ Android an iOS integration
+  /// also tracks this but what about Web, Linux, Windows and MacOS?
+  /// The message could include that this is a Flutter message.
```

Comment:
that's a good question, right now it's gonna make 'duplicated' breadcrumbs as the Native SDKs (only Android and iOS for now) already track App and Activity lifecycles and we don't officially support the other platforms.
Does the Flutter `WidgetsBindingObserver` API work for all the platforms?

Reply:
Yes, it does work on all platforms. `WidgetsBindingObserver` is the universal interface to get information about the hosts windows in which Flutter is embedded.